src/zh/happymh/src/eu/kanade/tachiyomi/extension/zh/happymh/Happymh.kt

```diff
@@ -151,22 +151,26 @@ class Happymh : HttpSource(), ConfigurableSource {
         return json.decodeFromString<ChapterListDto>(script).chapterList.map {
             SChapter.create().apply {
                 val chapterId = it.id
-                url = "/v2.0/apis/manga/read?code=$comicId&cid=$chapterId&v=v2.13"
+                url = "/reads/$comicId/$chapterId"
                 name = it.chapterName
             }
         }
     }
 
+    override fun getChapterUrl(chapter: SChapter): String {
+        return baseUrl + chapter.url
+    }
+
     // Pages
 
     override fun pageListRequest(chapter: SChapter): Request {
```

Comment:
throw exception saying to refresh chapter list if old format is detected

Reply:
In getChapterUrl() or pageListRequest(), or both?

Reply:
pageListRequest